src/pages/EntityDetails/Unit/Unit.tsx

```diff
@@ -38,23 +41,28 @@ export default function Unit() {
   const machines = useSelector(getModelMachines(modelUUID));
   const isK8s = useAppSelector((state) => isKubernetesModel(state, modelUUID));
 
+  const unit = useMemo(
+    () => (unitIdentifier && units?.[unitIdentifier]) || null,
+    [units, unitIdentifier],
+  );
```

Comment:
There's probably no value in a `useMemo` here (it might even be slower) as retrieving the unit will be very fast.